packages/cw-paginate/src/lib.rs

```diff
@@ -55,7 +57,19 @@ where
     K: Bounder<'a> + KeyDeserialize<Output = K> + 'static,
     V: serde::de::DeserializeOwned + serde::Serialize,
 {
-    let items = map.range(deps.storage, start_after.map(Bound::exclusive), None, order);
+    let mut range_min: Option<Bound<'a, K>> = None;
+    let mut range_max: Option<Bound<'a, K>> = None;
+
+    match order {
+        Order::Ascending => {
+            range_min = start_after.map(Bound::exclusive);
+        }
+        Order::Descending => {
+            range_max = start_after.map(Bound::exclusive);
+        }
+    }
```

Comment:
Great catch! Thank you! It is good perspective to know that there are bugs in this code. We are not infallible. :)

Small suggestion:

```suggestion
    let (range_min, range_max) = match order {
        Order::Ascending => {
            (start_after.map(Bound::exclusive), None)
        }
        Order::Descending => {
            (None, start_after.map(Bound::exclusive))
        }
    }
```

Reply:
applied suggestion